tests/saw/spec/handshake/handshake_io_lowlevel.saw

```diff
@@ -64,19 +64,24 @@ let config_cca_type config = (crucible_field config "client_cert_auth_type");
 let ocsp_status_size cert_and_key =
     crucible_field (crucible_field (cert_and_key) "ocsp_status") "size";
 
-// conn->config->use_tickets and conn->config->quic_enabled are the fields we
-// care about, but they are in a bitfield. We currently handle this by
-// initializing the entire bitfield with a symbolic value and imposing a
-// precondition that bit 3 (i.e., use_tickets) must be equal to 0.
-// Bit 9 (i.e., quic_enabled), on the other hand, is allowed to be either
-// 0 or 1.
-
-// It's also noteworthy that we use the crucible_elem here, which will
-// cause problems if the fields are reordered. This is reasonably unlikely
-// because it makes sense to have the bitfield at the start, but it is a bit
-// fragile nonetheless. If https://github.com/GaloisInc/saw-script/issues/1461
-// were implemented, we could reference the names of bitfields instead of their
-// offsets, which would avoid this issue.
+// At the moment, we care about three fields that are in bitfields:
+//
+// * conn->quic_enabled
+// * conn->config->use_tickets
+// * conn->config->quic_enabled
+//
+// We currently handle this by initializing the entirety of each bitfield with
+// a symbolic value and imposing appropriate preconditions on them. See the
+// comments in setup_connection_common next to conn_bitfield and
+// config_bitfield for more details about these preconditions.
+
+// It's also noteworthy that we use crucible_elem here, which will
+// cause problems if the fields are reordered. This is reasonably unlikely, but
+// it is a bit fragile nonetheless.
+// If https://github.com/GaloisInc/saw-script/issues/1461 were implemented, we
+// could reference the names of bitfields instead of their offsets, which would
+// avoid this issue.
+let conn_bitfield pconn = (crucible_elem pconn 80);
```

Comment:
This is not maintainable. We add fields to the connection fairly frequently. You're going to need to modify the C code to move all the flags to the beginning of the structure if you want to use crucible_elem in a reasonable way.

Reply:
OK. Currently, there are two bitfields within `s2n_connection`. There is this one:

https://github.com/aws/s2n-tls/blob/d5a044b8ce4eee8273397ee99d224faf8b562a39/tls/s2n_connection.h#L80-L111

And this one (which actually contains `quic_enabled`, the field we care about in the SAW specification):

https://github.com/aws/s2n-tls/blob/d5a044b8ce4eee8273397ee99d224faf8b562a39/tls/s2n_connection.h#L325-L354

Would you suggest moving both of these to be at the beginning of `s2n_connection` to form? Or just the latter?

Reply:
I'd prefer both.

Currently [corked_io is not modeled as a bitflag](https://github.com/aws/s2n-tls/blob/main/tests/saw/spec/handshake/handshake_io_lowlevel.saw#L38-L39), which is why the second bitfield exists-- we can't add any more flags to the first without breaking SAW. However, if you model corked_io as an actual bitflag, you'll also need to get [the advance message spec working](https://github.com/aws/s2n-tls/blob/main/tests/saw/spec/handshake/handshake_io_lowlevel.saw#L299) that way :\

Reply:
I've pushed a new version of this PR which should address these concerns:

* The first commit re-applies the changes from #3031, as they were reverted in #3077.
* The second commit moves all of the bitfields in `s2n_connection` to the front, per your suggestion.
* The third commit updates the SAW specifications to adapt to the new positions of the bitfields in `s2n_connection`.

The first commit is unchanged from #3031, so only the second and third commits are interesting from a reviewer perspective. Let me know what you think.